cellpack/autopack/Environment.py

```diff
@@ -999,51 +987,8 @@ def create_ingredient(self, recipe, arguments):
         if "place_method" not in arguments:
             arguments["place_method"] = self.place_method
         ingredient_type = arguments["type"]
-
-        if ingredient_type == INGREDIENT_TYPE.SINGLE_SPHERE:
-            radius = arguments["radius"]
-            arguments = IOutils.IOingredientTool.clean_arguments(
-                Ingredient.ARGUMENTS, **arguments
-            )
-            ingr = SingleSphereIngr(radius, **arguments)
-        elif ingredient_type == INGREDIENT_TYPE.MULTI_SPHERE:
-            arguments = IOutils.IOingredientTool.clean_arguments(
-                Ingredient.ARGUMENTS, **arguments
-            )
-            ingr = MultiSphereIngr(**arguments)
-        elif ingredient_type == INGREDIENT_TYPE.MULTI_CYLINDER:
-            arguments = IOutils.IOingredientTool.clean_arguments(
-                Ingredient.ARGUMENTS, **arguments
-            )
-            ingr = MultiCylindersIngr(**arguments)
-        elif ingredient_type == INGREDIENT_TYPE.SINGLE_CUBE:
-            bounds = arguments["bounds"]
-            arguments = IOutils.IOingredientTool.clean_arguments(
-                Ingredient.ARGUMENTS, **arguments
-            )
-            ingr = SingleCubeIngr(bounds, **arguments)
-        elif ingredient_type == INGREDIENT_TYPE.SINGLE_CYLINDER:
-            arguments = IOutils.IOingredientTool.clean_arguments(
-                Ingredient.ARGUMENTS, **arguments
-            )
-            ingr = SingleCylinderIngr(**arguments)
-        elif ingredient_type == INGREDIENT_TYPE.GROW:
-            arguments = IOutils.IOingredientTool.clean_arguments(
-                GrowIngredient.ARGUMENTS, **arguments
-            )
-            ingr = GrowIngredient(**arguments)
-        elif ingredient_type == INGREDIENT_TYPE.GROW:
-            arguments = IOutils.IOingredientTool.clean_arguments(
-                GrowIngredient.ARGUMENTS, **arguments
-            )
-            ingr = ActinIngredient(**arguments)
-        else:
-            radius = arguments["radius"]
-            arguments = IOutils.IOingredientTool.clean_arguments(
-                Ingredient.ARGUMENTS, **arguments
-            )
-            arguments["type"] = INGREDIENT_TYPE.SINGLE_SPHERE
-            ingr = SingleSphereIngr(radius, **arguments)
+        ingredient_class = ingredient.get_ingredient_class(ingredient_type)
+        ingr = ingredient_class(**arguments)
```

Comment:
Assuming the `clean_arguments` step happens in the ingredient classes?

Reply:
no, it's now that we're getting much more standard arguments from the loader so we don't need to do this extra step anymore. This was a stop gap while we were removing some old attributes in old recipes